pkg/tsdb/loki/loki.go

```diff
@@ -106,17 +82,7 @@ func (s *Service) QueryData(ctx context.Context, req *backend.QueryDataRequest)
 		return result, err
 	}
 
-	client := &client.DefaultClient{
-		Address:  dsInfo.URL,
-		Username: dsInfo.BasicAuthUser,
-		Password: dsInfo.BasicAuthPassword,
-		TLSConfig: config.TLSConfig{
-			InsecureSkipVerify: dsInfo.TLSClientConfig.InsecureSkipVerify,
-		},
-		Tripperware: func(t http.RoundTripper) http.RoundTripper {
-			return dsInfo.HTTPClient.Transport
-		},
-	}
+	api := newLokiAPI(dsInfo.HTTPClient, dsInfo.URL, s.plog)
```

Comment:
Nit. Since this doesn't depend on the query arguments you can actually create and store the `api` in `dsInfo` instead of `http.Client`.

Reply:
thanks for the idea, i will keep it in mind 👍  (i need to make sure i can easily test the code, so this might make it harder, we'll see)